model/project_ref.go

```diff
@@ -2591,9 +2599,15 @@ func ValidateContainers(pRef *ProjectRef, containers []Container) error {
 		}
 		catcher.ErrorfWhen(container.Size != "" && !ok, "size '%s' is not defined anywhere", container.Size)
 		if container.Credential != "" {
-			cs, ok := pRef.ContainerSecrets[container.Credential]
-			catcher.ErrorfWhen(!ok, "credential '%s' is not defined anywhere", container.Credential)
-			catcher.ErrorfWhen(ok && cs.Type != ContainerSecretRepoCred, "container credential named '%s' exists but is not valid for use as a repository credential", container.Credential)
+			var matchingSecret *ContainerSecret
```

Comment:
I know it shouldn't be possible due to the upserting/updating logic we have but since `Name` is effectively the key for the secret it might be worth ensuring we don't have the same name twice in the list of secrets

Reply:
If there's a duplicate in the project ref, it's probably too late to validate it here because this block is for validating the project YAML after the project ref has already been updated. Supposedly, the logic in `ValidateContainerSecrets` should already consolidate duplicates before we save the project ref, so we wouldn't need to check it here.

Reply:
Ok sgtm